ksql-engine/src/main/java/io/confluent/ksql/analyzer/Analyzer.java

```diff
@@ -313,26 +313,19 @@ private void throwOnUnknownColumnReference() {
           new ExpressionAnalyzer(analysis.getFromSourceSchemas());
 
       for (final SelectExpression selectExpression : analysis.getSelectExpressions()) {
-        expressionAnalyzer.analyzeExpression(selectExpression.getExpression(), false);
+        expressionAnalyzer.analyzeExpression(selectExpression.getExpression());
       }
 
       analysis.getWhereExpression().ifPresent(where -> {
-        final boolean allowWindowMetaFields = pullQuery
-            && analysis.getFromDataSources().get(0)
-            .getDataSource()
-            .getKsqlTopic()
-            .getKeyFormat()
-            .isWindowed();
```

Comment:
We can now remove old special handling of window bounds columns for pull queries.